File: test/language/expressions/java/expressions/ParameterTypeTest.java
package expressions;

import static org.junit.Assert.*;

import org.junit.Test;

import expressions.parameter_type.Color;
import expressions.parameter_type.ParameterTypeExpression;

public class ParameterTypeTest
{
    @Test
    public void bitSizeOfWithOptional()
    {
        final ParameterTypeExpression parameterTypeExpression = new ParameterTypeExpression(Color.RED,
                PARAMETER_TYPE_EXPRESSION_VALUE, true);

        assertEquals(PARAMETER_TYPE_EXPRESSION_BIT_SIZE_WITH_OPTIONAL, parameterTypeExpression.bitSizeOf());
    }

    @Test
    public void bitSizeOfWithoutOptional()
    {
        final ParameterTypeExpression parameterTypeExpression = new ParameterTypeExpression(Color.BLUE);
        parameterTypeExpression.setValue(PARAMETER_TYPE_EXPRESSION_VALUE);

        assertEquals(PARAMETER_TYPE_EXPRESSION_BIT_SIZE_WITHOUT_OPTIONAL,
                parameterTypeExpression.bitSizeOf());
    }

    private static final byte PARAMETER_TYPE_EXPRESSION_VALUE = (byte)0x7B;

    private static final int PARAMETER_TYPE_EXPRESSION_BIT_SIZE_WITH_OPTIONAL = 8;
    private static final int PARAMETER_TYPE_EXPRESSION_BIT_SIZE_WITHOUT_OPTIONAL = 7;
}

File: test/language/expressions/cpp98/ParameterTypeTest.cpp
#include "math.h"

#include "gtest/gtest.h"

#include "expressions/parameter_type/ParameterTypeExpression.h"

namespace expressions
{
namespace parameter_type
{

TEST(ParameterTypeTest, bitSizeOfWithOptional)
{
    ParameterTypeExpression parameterTypeExpression;
    parameterTypeExpression.initialize(Color::RED);
    parameterTypeExpression.setValue(0x7B);
    parameterTypeExpression.setIsParameterRed(true);

    const size_t parameterTypeExpressionBitSizeWithOptional = 8;
    ASSERT_EQ(parameterTypeExpressionBitSizeWithOptional, parameterTypeExpression.bitSizeOf());
}

TEST(ParameterTypeTest, bitSizeOfWithoutOptional)
{
    ParameterTypeExpression parameterTypeExpression;
    parameterTypeExpression.initialize(Color::BLUE);
    parameterTypeExpression.setValue(0x7A);

    const size_t parameterTypeExpressionBitSizeWithoutOptional = 7;
    ASSERT_EQ(parameterTypeExpressionBitSizeWithoutOptional, parameterTypeExpression.bitSizeOf());
}

} // namespace parameter_type
} // namespace expressions
